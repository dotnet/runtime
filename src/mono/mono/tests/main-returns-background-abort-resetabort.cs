
using System;
using System.Threading;

public class foo {
	public static void Main() {
		Thread thr=new Thread(new ThreadStart(foo.thread));
		thr.IsBackground=true;
		thr.Start();
		Thread.Sleep(600);
		Console.WriteLine("Aborting child thread");
		thr.Abort();
		Thread.Sleep(600);
		Console.WriteLine("Main thread returns");
	}

	public static void thread() {
		try {
			Console.WriteLine("Thread running");
			Thread.Sleep(500);
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset!");
		} finally {
			Console.WriteLine("ThreadAbortException finally");
		}
		try {
			Console.WriteLine("Thread running");
			Thread.Sleep(500);
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset!");
		} finally {
			Console.WriteLine("ThreadAbortException finally");
		}
		try {
			Console.WriteLine("Thread running");
			Thread.Sleep(500);
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset!");
		} finally {
			Console.WriteLine("ThreadAbortException finally");
		}
		try {
			Console.WriteLine("Thread running");
			Thread.Sleep(500);
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset!");
		} finally {
			Console.WriteLine("ThreadAbortException finally");
		}
		try {
			Console.WriteLine("Thread running");
		} catch(ThreadAbortException) {
			Thread.ResetAbort();
			Console.WriteLine("Abort reset!");
		} finally {
			Console.WriteLine("ThreadAbortException finally");
		}
	}
}

