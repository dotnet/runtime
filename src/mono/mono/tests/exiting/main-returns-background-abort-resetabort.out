Thread running
ThreadAbortException finally
Thread running
Aborting child thread
Abort reset!
ThreadAbortException finally
Thread running
ThreadAbortException finally
Thread running
Main thread returns
